nntrainer/src/pooling2d_layer.cpp

```diff
@@ -23,7 +23,25 @@ namespace nntrainer {
 
 int Pooling2DLayer::initialize(bool last) {
   int status = ML_ERROR_NONE;
-  // NYI
+  if (input_dim.getDataLen() == 1) {
+    ml_logw("Warning: the length of previous layer dimension is one");
+  }
+  if (input_dim.batch() <= 0 || input_dim.height() <= 0 ||
+      input_dim.width() <= 0 || input_dim.channel() <= 0) {
+    ml_loge("Error: Dimension must be greater than 0");
+    return ML_ERROR_INVALID_PARAMETER;
+  }
+
+  this->last_layer = last;
+  output_dim.batch(input_dim.batch());
+  output_dim.channel(input_dim.channel());
+  output_dim.height(
+    (input_dim.height() - pooling_size[0] + 2 * padding[0]) / stride[0] + 1);
```

Comment:
IMO, `void TensorDim::setTensorDim(unsigned int idx, unsigned int value)` should validate that `input > 0` from now on

Reply:
I think the validation of this ``` if (value <=0) ``` might be done at TensorDim::setTensorDim(unsigned int idx, unsigned int value). and the type of value is unsigned int. ( It causes runtime error I guess ).

Reply:
Right. Thanks for that :)